daemon/containerd/service.go

```diff
@@ -201,3 +203,10 @@ func (i *ImageService) GetContainerLayerSize(ctx context.Context, containerID st
 	// TODO(thaJeztah): include content-store size for the image (similar to "GET /images/json")
 	return rwLayerUsage.Size, rwLayerUsage.Size + unpackedUsage.Size, nil
 }
+
+func (i *ImageService) hostPlatform() platforms.MatchComparer {
+	if i.defaultPlatformOverride != nil {
+		return i.defaultPlatformOverride
+	}
+	return platforms.Default()
+}
```

Comment:
As this is only used in a single location (do we expect more?) perhaps inline this code and add a comment to explain that the override would only be set when testing. It's probably better to have it very "in your face" that that's what we do and not make it look nice 😄

Reply:
Inlined